euphonic/qpoint_frequencies.py

```diff
@@ -168,6 +188,17 @@ def _calculate_dos(self, dos_bins: Quantity,
                     pdf = norm.pdf(dos_bins_calc, loc=freqs[q,m],
                                    scale=mode_widths[q,m])
                     dos += pdf*weights[q]*mode_weights_calc[q, m]
+        elif mode_widths is not None and adaptive_method=='fast':
+            # fast, approximate method for adaptive broadening
+            freqs = np.ravel(freqs)
+            mode_widths = np.ravel(mode_widths).to('hartree').magnitude
```

Comment:
It looks a bit strange (and scary) that this one method requires the input data to be cast to particular units. Would it hurt performance to put that inside `fast_broaden()`?

Reply:
Yeah fair point, I'll move these lines inside of the function